packages/events/src/router/event/index.ts

```diff
@@ -295,8 +298,58 @@ export const eventRouter = router({
       return getIndexedEvents(userId)
     }),
   search: publicProcedure
+    .meta({
+      openapi: {
+        summary: 'Search for events',
+        method: 'GET',
+        tags: ['Search'],
+        path: '/events/search'
+      }
+    })
     .use(requiresAnyOfScopes(CONFIG_SEARCH_ALLOWED_SCOPES))
-    .input(QueryType)
+    .input(
+      z.object({
+        type: z.literal('and').or(z.literal('or')).openapi({ default: 'and' }),
+        clauses: z.preprocess(
+          (val) => {
+            if (typeof val === 'string') {
+              return [JSON.parse(val)]
+            }
+            if (Array.isArray(val)) {
+              return val.map((v) => (typeof v === 'string' ? JSON.parse(v) : v))
+            }
+            return val
+          },
+          z.array(QueryExpression).openapi({
+            default: [
+              {
+                eventType: 'tennis-club-membership',
+                status: {
+                  type: 'anyOf',
+                  terms: [
+                    'CREATED',
+                    'NOTIFIED',
+                    'DECLARED',
+                    'VALIDATED',
+                    'REGISTERED',
+                    'CERTIFIED',
+                    'REJECTED',
+                    'ARCHIVED'
+                  ]
+                },
+                updatedAt: {
+                  type: 'range',
+                  gte: '2025-05-22',
+                  lte: '2025-05-29'
+                },
+                data: {}
+              }
+            ]
+          })
+        )
+      })
```

Comment:
Let's make this into a constant and hide it in some other file, we should aim to keep this router file rather clean :)